--- board-omap4panda.c.old	2012-01-31 10:50:25.532288027 +0800
+++ board-omap4panda.c	2012-01-30 15:20:02.136522196 +0800
@@ -319,7 +319,7 @@
 
+static struct i2c_board_info __initdata panda_i2c_boardinfo[] = {
+#if (defined(CONFIG_SENSORS_DMARD08) || defined(CONFIG_SENSORS_DMARD08_MODULE))
+	{
+		I2C_BOARD_INFO("dmt", 0x1C),
+	},
+#endif
+};
+
 static int __init omap4_panda_i2c_init(void)
 {
 	omap4_pmic_init("twl6030", &omap4_panda_twldata);
-	omap_register_i2c_bus(2, 400, NULL, 0);
+	omap_register_i2c_bus(2, 400, panda_i2c_boardinfo, ARRAY_SIZE(panda_i2c_boardinfo));
 	/*
 	 * Bus 3 is attached to the DVI port where devices like the pico DLP
 	 * projector don't work reliably with 400kHz
